tensorflow-examples/src/main/java/org/tensorflow/model/examples/cnn/fastrcnn/FasterRcnnInception.java

```diff
@@ -0,0 +1,342 @@
+            Map<String, Tensor> outputTensorMap = model.function("serving_default").call(feedDict);
+            //detection_classes is a model output name
+            TFloat32 detectionClasses = (TFloat32) outputTensorMap.get("detection_classes");
+            TFloat32 detectionBoxes = (TFloat32) outputTensorMap.get("detection_boxes");
+            TFloat32 numDetections = (TFloat32) outputTensorMap.get("num_detections");
+            TFloat32 detectionScores = (TFloat32) outputTensorMap.get("detection_scores");
+
+            int numDetects = (int) numDetections.getFloat(0);
+            if (numDetects > 0) {
+                try {
+                    BufferedImage bufferedImage = ImageIO.read(new File(imagePath));
+                    Graphics2D graphics2D = bufferedImage.createGraphics();
+
+                    //TODO tf.image.combinedNonMaxSuppression
+                    for (int n = 0; n < numDetects; n++) {
+                        //put probability and position in outputMap
+                        float detectionScore = detectionScores.getFloat(0, n);
+                        //only include those classes with detection score greater than 0.3f
+                        if (detectionScore > 0.3f) {
+                            float classVal = detectionClasses.getFloat(0, n);
+                            //TODO tf.image.drawBoundingBoxes
+                            int x1 = (int) (shapeArray[1] * detectionBoxes.getFloat(0, n, 1));
+                            int y1 = (int) (shapeArray[0] * detectionBoxes.getFloat(0, n, 0));
+                            int x2 = (int) (shapeArray[1] * detectionBoxes.getFloat(0, n, 3));
+                            int y2 = (int) (shapeArray[0] * detectionBoxes.getFloat(0, n, 2));
+                            graphics2D.setPaint(Color.RED);
+                            graphics2D.setStroke(new BasicStroke(5));
+                            graphics2D.drawRect(x1, y1, x2 - x1, y2 - y1);
+                            graphics2D.setPaint(Color.BLACK);
+                            //add a label with percentage score
+                            graphics2D.drawString(cocoTreeMap.get(classVal - 1) + " " +
+                                    (new DecimalFormat("#.##").format(detectionScore * 100)) +
+                                    "%", x1, y1);
+                        }
+                    }
+                    //TODO tf.image.encodeJpeg
+                    ImageIO.write(bufferedImage, "jpg", new File(outputImagePath));
+                } catch (IOException e) {
+                    System.err.println("Exception with writing image " + e.getMessage());
+                }
+            }
+            //close all tensors in outputTensorMap
+            for (String key : outputTensorMap.keySet()){
+                Tensor tensor = outputTensorMap.get(key);
+                tensor.close();
+            }
+
+            reshapeTensor.close();
+            outputImage.close();
```

Comment:
Since this is a tutorial, I think we should show the best approach for closing tensors, which is with `try-with-resources`. I saw that in a previous commit, you were encapsulating each tensors resulting from the function call in such block, i.e.
```java
            //detection_classes is a model output name
            try (TFloat32 detectionClasses = (TFloat32) outputTensorMap.get("detection_classes");
                 TFloat32 detectionBoxes = (TFloat32) outputTensorMap.get("detection_boxes");
                 TFloat32 numDetections = (TFloat32) outputTensorMap.get("num_detections");
                 TFloat32 detectionScores = (TFloat32) outputTensorMap.get("detection_scores")) {
```
I'd suggest you revert to that. `reshapeTensor` and `outputImage` should also be encapsulated, e.g.
```java
try (TUint8 reshapeTensor = (TUint8) runner.fetch(reshape).run().get(0)) { 
   ... 
}
```

Even what I'm proposing here does not guarantee that all resources will be closed (since we assume that we fetch all tensors from a session run or a function call), but it is better than closing explicitly like in the current example, which will never happen if an exception in thrown before (so someone copy&pasting that example in a Java function could endup with memory leaks). 

We are actually working on a better solution for closing all resources properly without all this headache.

Reply:
I've made sure that all the tensors created by the model function call are encapsulated in the try block so hopefully they will all be closed. I've also encapsulated the other two tensors.